packages/atomic/src/utils/tab-utils.spec.ts

```diff
@@ -0,0 +1,112 @@
+import {getActiveTab, shouldDisplayOnCurrentTab} from './tab-utils';
+
+describe('getActiveTab', () => {
+  it('returns null when state.tabSet is undefined', () => {
+    const state = {};
+    expect(getActiveTab(state)).toEqual(null);
+  });
+
+  it('returns null when no tab is active', () => {
+    const state = {
+      tabSet: {
+        tab1: {id: 'tab1', isActive: false, expression: ''},
+        tab2: {id: 'tab2', isActive: false, expression: ''},
+        tab3: {id: 'tab3', isActive: false, expression: ''},
+      },
+    };
+    expect(getActiveTab(state)).toEqual(null);
+  });
+
+  it('returns the active tab when one tab is active', () => {
+    const state = {
+      tabSet: {
+        tab1: {id: 'tab1', isActive: false, expression: ''},
+        tab2: {id: 'tab2', isActive: true, expression: ''},
+        tab3: {id: 'tab3', isActive: false, expression: ''},
+      },
+    };
+    expect(getActiveTab(state)).toEqual({tab: 'tab2'});
+  });
+
+  it('returns the first active tab when multiple tabs are active', () => {
```

Comment:
Mostly musing here but...

If knowing what tab is currently active is important for facets (and other controllers/components), shouldn't we expose the id of the active tab on the state of those controllers? It would either be undefined (if the tab section isn't in the state at all), null (if the tab section is there but no tab is selected) or a string (the id of the currently selected tab).

Maybe the shape of the tab section could also change in the engine state so that instead of being a `Record<string, TabSlice>` it would be something like

```typescript
export type TabSetState = { tabs: Record<string, TabSlice>; activeTab?: string }
```

and then when we dispatch `updateActiveTab(id)`, in the tabSetReducer we'd also set the active tab there... which means we wouldn't have to iterate through all tabs all the time to find the active one.

Now the problem with what I'm suggesting above is that modifying the definition of the `TabSetState` type would definitely be a breaking change.

Another possibility would be to add a new section in the state just to represent the active tab, and perhaps modify the TabSetState type in v3 or v4... but maybe in v4 the engine's state won't be public anymore and then we can do whatever we want ;)

Reply:
> Now the problem with what I'm suggesting above is that modifying the definition of the TabSetState type would definitely be a breaking change.

That's the problem ;)